Source/Core/GroundPolylineGeometry.js

```diff
@@ -0,0 +1,1054 @@
+            cartographicsArray.push(vertexCartographic.longitude);
+
+            interpolateSegment(cartographics[i], cartographics[i + 1], minHeight, maxHeight, granularity, ellipsoid, normalsArray, bottomPositionsArray, topPositionsArray, cartographicsArray);
+        }
+
+        // Last point - either loop or attach a normal "perpendicular" to the wall.
+        var endCartographic = cartographics[cartographicsLength - 1];
+        var preEndCartographic = cartographics[cartographicsLength - 2];
+
+        vertexBottom = getPosition(ellipsoid, endCartographic, minHeight, vertexBottom);
+        vertexTop = getPosition(ellipsoid, endCartographic, maxHeight, vertexTop);
+
+        if (loop) {
+            var postEndCartographic = cartographics[0];
+            previousBottom = getPosition(ellipsoid, preEndCartographic, minHeight, previousBottom);
+            nextBottom = getPosition(ellipsoid, postEndCartographic, minHeight, nextBottom);
+
+            vertexNormal = computeVertexMiterNormal(previousBottom, vertexBottom, vertexTop, nextBottom, vertexNormal);
+        } else {
+            vertexNormal = computeRightNormal(preEndCartographic, endCartographic, maxHeight, ellipsoid, vertexNormal);
+        }
+
+        index = normalsArray.length;
+        Cartesian3.pack(vertexNormal, normalsArray, index);
+        Cartesian3.pack(vertexBottom, bottomPositionsArray, index);
+        Cartesian3.pack(vertexTop, topPositionsArray, index);
+        cartographicsArray.push(endCartographic.latitude);
+        cartographicsArray.push(endCartographic.longitude);
+
+        if (loop) {
+            interpolateSegment(endCartographic, startCartographic, minHeight, maxHeight, granularity, ellipsoid, normalsArray, bottomPositionsArray, topPositionsArray, cartographicsArray);
+            index = normalsArray.length;
+            // Copy the first vertex
+            for (i = 0; i < 3; ++i) {
+                normalsArray[index + i] = normalsArray[i];
+                bottomPositionsArray[index + i] = bottomPositionsArray[i];
+                topPositionsArray[index + i] = topPositionsArray[i];
+            }
+            cartographicsArray.push(startCartographic.latitude);
+            cartographicsArray.push(startCartographic.longitude);
+        }
+
+        return generateGeometryAttributes(loop, projection, bottomPositionsArray, topPositionsArray, normalsArray, cartographicsArray);
+    };
+
+    // If the end normal angle is too steep compared to the direction of the line segment,
+    // "break" the miter by rotating the normal 90 degrees around the "up" direction at the point
+    // For ultra precision we would want to project into a plane, but in practice this is sufficient.
+    var lineDirectionScratch = new Cartesian3();
+    var matrix3Scratch = new Matrix3();
+    var quaternionScratch = new Quaternion();
+    function breakMiter(endGeometryNormal, startBottom, endBottom, endTop) {
```

Comment:
@bagnell can any of this miter logic be **reasonably** shared with the other polyline geometry implementation?

Reply:
No, the miter logic for other polylines is in the vertex shader. I discussed this with @likangning93. We would need so much information that we ran out of vertex attributes.